infra/app/function.bicep

```diff
@@ -0,0 +1,90 @@
+param name string
+param location string = ''
+param appServicePlanId string 
+param storageAccountName string = ''
+param tags object = {}
+
+param appSettings array = []
+param serviceName string = 'Function'
+param applicationInsightsName string = ''
+param runtimeName string
+param runtimeVersion string
+param ClientKey string
+param AzureOpenAIName string = ''
+param AzureCognitiveSearchName string = ''
+param FormRecognizerName string = ''
+param ContentSafetyName string = ''
+
+
+module Function '../core/host/functions.bicep' = {
+  name: '${name}-app-module'
+  params: {
+    name: name
+    location: location
+    tags: union(tags, { 'azd-service-name': serviceName })
+    appServicePlanId: appServicePlanId
+    storageAccountName: storageAccountName
+    runtimeName: runtimeName
+    runtimeVersion: runtimeVersion
+    appSettings: union(toObject(appSettings, entry => entry.name, entry => entry.value), {
+      AzureWebJobsStorage: 'DefaultEndpointsProtocol=https;AccountName=${storage.name};AccountKey=${storage.listKeys().keys[0].value};EndpointSuffix=${environment().suffixes.storage}'
+      AZURE_BLOB_ACCOUNT_KEY: storage.listKeys().keys[0].value
+      APPINSIGHTS_INSTRUMENTATIONKEY: applicationInsights.properties.InstrumentationKey
+      AZURE_OPENAI_KEY: openai.listKeys().key1
+      AZURE_SEARCH_KEY: search.listAdminKeys().primaryKey
+      AZURE_FORM_RECOGNIZER_KEY: formrecognizer.listKeys().key1
+      AZURE_CONTENT_SAFETY_KEY: ContentSafety.listKeys().key1
+    })
+  }
+}
+
+resource storage 'Microsoft.Storage/storageAccounts@2021-09-01' existing = {
```

Comment:
Each one of these "exisiting" calls is a costly ARM deployment.  So we want to minimize them.  Instead of this, you should be passing the keys in via app settings from main.bicep.

Reply:
If this is done, the current way I can think of is to pass these `keys` as `output` from the bicep file in the module into `main.bicep`, which doesn‘t seem very secure. Any better suggestions? Please take a look at my [latest commit](https://github.com/jongio/chat-with-your-data-solution-accelerator/pull/2/commits/c00f55a8f5f9c9b4f1aa6318fe493ef3a31cf995#diff-495e118ae4bb45392efbfa5a2f99260e3e2f72da47e6fdfe73d929d5f07c6068R64). Is this okay?